### ISSUE TITLE
NPE when validating a null query parameter

### ISSUE THREAD

**Issue by Dimitris-Christodoulou** [openapi4j/openapi4j#146]
**Version**  
1.0.2  

**Describe the bug**  
A `NullPointerException` is thrown when validating a request with a `null` query parameter.

**Expected behavior**  
Expected a validation error with a message saying the query parameter cannot be `null`.

**Additional context**  
I have tracked down the issue to [this](https://github.com/openapi4j/openapi4j/blob/1d089daa4793eba4fcf55aeb0f4c0cb4f744bbb5/openapi-operation-validator/src/main/java/org/openapi4j/operation/validator/util/convert/style/FormStyleConverter.java#L138) line of code: 
```java
      paramValues.stream().findFirst().orElse(null));
```

`findFirst()` casts the result to `Optional<T>` so it [throws](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#findFirst--) a `NullPointerException` if the result is `null`. To avoid the issue we could filter out any null values, e.g.
```java
   paramValues.stream().filter(Objects::nonNull).findFirst().orElse(null));
```
**To Reproduce**  
Please run the Java program below.

**Reproducer**  
```java
import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.util.Collections;
import org.openapi4j.core.exception.ResolutionException;
import org.openapi4j.core.validation.ValidationException;
import org.openapi4j.operation.validator.model.Request;
import org.openapi4j.operation.validator.model.Request.Method;
import org.openapi4j.operation.validator.model.impl.DefaultRequest;
import org.openapi4j.operation.validator.validation.OperationValidator;
import org.openapi4j.parser.OpenApi3Parser;
import org.openapi4j.parser.model.v3.OpenApi3;
import org.openapi4j.parser.model.v3.Operation;
import org.openapi4j.parser.model.v3.Path;
import org.openapi4j.schema.validator.ValidationData;

public class QueryParamTest {

  public static void main(final String[] args)
      throws IOException, ResolutionException, ValidationException {
    final String apiDefinition =
        "openapi: 3.0.0\n"
            + "info:\n"
            + "  title: Sample API\n"
            + "  description: A sample API to be used in unit-tests\n"
            + "  version: v0\n"
            + "paths:\n"
            + "  /users:\n"
            + "    get:\n"
            + "      operationId: GetUser\n"
            + "      parameters:\n"
            + "        - name: userID\n"
            + "          in: query\n"
            + "          required: false\n"
            + "          schema:\n"
            + "            $ref: '#/components/schemas/UserID'\n"
            + "      responses:\n"
            + "        '200':\n"
            + "          description: The user\n"
            + "          content:\n"
            + "            application/json:\n"
            + "              schema:\n"
            + "                $ref: '#/components/schemas/User'\n"
            + "\n"
            + "components:\n"
            + "  schemas:\n"
            + "    User:\n"
            + "      type: string\n"
            + "\n"
            + "    UserID:\n"
            + "      type: integer\n"
            + "      format: int64\n"
            + "\n";

    final File yaml =
        Files.write(Files.createTempFile("openapi", ".yaml"), apiDefinition.getBytes()).toFile();
    final DefaultRequest.Builder builder = new DefaultRequest.Builder("/users", Method.GET);

    builder.body(null);
    builder.cookies(Collections.emptyMap());
    builder.headers(Collections.emptyMap());

    builder.query("userID");

    final Request request = builder.build();

    final OpenApi3 openApi3 = new OpenApi3Parser().parse(yaml, false);
    final Path path = openApi3.getPathItemByOperationId("GetUser");
    final Operation operation = openApi3.getOperationById("GetUser");

    final OperationValidator validator = new OperationValidator(openApi3, path, operation);
    final ValidationData<Void> data = new ValidationData<>();
    validator.validateQuery(request, data);
    assert data.isValid();
  }
}
```

**Comment by llfbandit** [comment]
Hi,

Thanks for the report. Fix will be available soon in 1.0.4-SNAPSHOT.
Are you able to validate?

**Comment by Dimitris-Christodoulou** [comment]
> Hi,
> 
> Thanks for the report. Fix will be available soon in 1.0.4-SNAPSHOT.
> Are you able to validate?

Thank you - I will try to validate this week

**Comment by Dimitris-Christodoulou** [comment]
@llfbandit Validated. Sorry for the delay.
